### ISSUE TITLE
Add message arguments to `declaration-property-unit-allowed-list`

### ISSUE THREAD

**Issue by glen-84** [stylelint/stylelint#6454]
### What is the problem you're trying to solve?

Setting a custom message for the `declaration-property-unit-allowed-list` rule.

### What solution would you like to see?

```js
"declaration-property-unit-allowed-list": [
    {
        "line-height": []
    },
    {
        message: (propertyName, _units) => {
            switch (propertyName) {
                case "line-height":
                    return "Prefer unitless numbers for line-height values";
            }
        }
    }
]
```

**Comment by ybiquitous** [comment]
@glen-84  Thanks for the proposal. See also <https://github.com/stylelint/stylelint/issues/6453#issuecomment-1304954997>.

I've labeled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

**Comment by glen-84** [comment]
@ybiquitous

I think that this issue was accidentally closed. This is for `allowed`, not `disallowed`. 🙂

/cc @chloerice.

**Comment by ybiquitous** [comment]
@glen-84 Oh, thank you for letting us know!
